airbyte-db/db-lib/src/main/java/io/airbyte/db/util/SSLCertificateUtils.java

```diff
@@ -0,0 +1,160 @@
+  }
+
+  public static URI keyStoreFromClientCertificate(
+                                                  final Certificate cert,
+                                                  final PrivateKey key,
+                                                  final String keyStorePassword,
+                                                  final FileSystem filesystem,
+                                                  final String directory)
+      throws KeyStoreException, CertificateException, IOException, NoSuchAlgorithmException {
+    final KeyStore keyStore = KeyStore.getInstance(PKCS_12);
+    keyStore.load(null);
+    keyStore.setKeyEntry(KEYSTORE_ENTRY_PREFIX, key, keyStorePassword.toCharArray(), new Certificate[] {cert});
+    return saveKeyStoreToFile(keyStore, keyStorePassword, filesystem, directory);
+  }
+
+  public static URI keyStoreFromClientCertificate(
+                                                  final String certString,
+                                                  final String keyString,
+                                                  final String keyStorePassword,
+                                                  final FileSystem filesystem,
+                                                  final String directory)
+      throws IOException, InterruptedException, NoSuchAlgorithmException, InvalidKeySpecException, CertificateException, KeyStoreException {
+
+    // Convert RSA key (PKCS#1) to PKCS#8 key
+    // Note: java.security doesn't have a built in support of PKCS#1 format. A conversion using openssl
+    // is necessary.
+    // Since this is a single operation it's better than adding an external lib (e.g BouncyCastle)
+    final Path tmpDir = Files.createTempDirectory(null);
+    final Path pkcs1Key = Files.createTempFile(tmpDir, null, null);
+    final Path pkcs8Key = Files.createTempFile(tmpDir, null, null);
+    pkcs1Key.toFile().deleteOnExit();
+    pkcs8Key.toFile().deleteOnExit();
+
+    Files.write(pkcs1Key, keyString.getBytes(StandardCharsets.UTF_8));
+    runProcess(
+        "openssl pkcs8 -topk8 -inform PEM -outform DER -in " + pkcs1Key.toAbsolutePath() + " -out " + pkcs8Key.toAbsolutePath()
+            + " -nocrypt -passout pass:" + keyStorePassword,
+        Runtime.getRuntime());
+
+    final PKCS8EncodedKeySpec spec = new PKCS8EncodedKeySpec(Files.readAllBytes(pkcs8Key));
+    PrivateKey privateKey;
+    try {
+      privateKey = KeyFactory.getInstance("RSA").generatePrivate(spec);
+    } catch (final InvalidKeySpecException ex1) {
+      try {
+        privateKey = KeyFactory.getInstance("DSA").generatePrivate(spec);
+      } catch (final InvalidKeySpecException ex2) {
+        privateKey = KeyFactory.getInstance("EC").generatePrivate(spec);
+      }
+    }
+
+    return keyStoreFromClientCertificate(fromPEMString(certString), privateKey, keyStorePassword, filesystem, directory);
+
+  }
+
+  public static URI keyStoreFromClientCertificate(
```

Comment:
this method appears to be unused - why do we need it?

Reply:
I created these utilities allowing the caller to provide a `FileSystem` so private keys can be saved on something other than a file. 
This is a convenience method that defaults to regular FileSystem–
I'm changing the caller to use it as I ended up not using a in memory FS I had in mind after discovering that the postgres connector cannot support it